query-engine/request-handlers/src/handler.rs

```diff
@@ -258,6 +259,10 @@ impl<'a> RequestHandler<'a> {
                 Some(t1) => Self::compare_values(t1, t2),
                 None => left == right,
             },
+            (ArgumentValue::Scalar(PrismaValue::Float(s1)), ArgumentValue::Scalar(PrismaValue::String(s2)))
+            | (ArgumentValue::Scalar(PrismaValue::String(s2)), ArgumentValue::Scalar(PrismaValue::Float(s1))) => {
+                BigDecimal::from_str(&s2).map(|s2| s2 == *s1).unwrap_or(false)
+            }
```

Comment:
I'm a bit worried that we encounter some kind of precision loss which would make the comparison fail, but I've got no better solution. Let's see!